internal/controllers/topology/cluster/current_state.go

```diff
@@ -272,6 +281,108 @@ func (r *Reconciler) getCurrentMachineDeploymentState(ctx context.Context, bluep
 	return state, nil
 }
 
+// getCurrentMachinePoolState queries for all MachinePools and filters them for their linked Cluster and
```

Comment:
Please re-diff MD and MP in this file, we made at least a performance optimization for MD's

Reply:
If the code is very similar, I wonder if there is a bigger potential for code reuse so they don't diverge again in the future? Is there any common logic that can be extracted and shared? maybe using go generics for the types?

Reply:
I reviewed [`getCurrentMachineDeploymentState`](https://github.com/kubernetes-sigs/cluster-api/blob/main/internal/controllers/topology/cluster/current_state.go#L160) and it's very similar to `getCurrentMachinePoolState` minus the MachineHealthCheck stuff. Am I missing one of the updates?